docs/rules/no-document-during-ssr.md

```diff
@@ -0,0 +1,34 @@
+# Disallow references to document during SSR (`lwc/no-document-during-ssr`)
+
+💼 This rule is enabled in the ✅ `recommended` config.
```

Comment:
I wouldn't make SSR rules recommended for now. Not all components should be SSR-ready, and those rules aren't fully stable yet.

Reply:
Agreed. I suspect we need a signal for many tools (eg IDE, builders) to understand whether a component is usable in an SSR context

Reply:
The complexity here is that ESLint operates on each file independently. It is not recommended to load another file to determine the behavior or a rule as it prevents ESLint from caching the lint report. 

One option would be to create a new [config](https://github.com/salesforce/eslint-config-lwc#configurations) in `@salesforce/eslint-config-lwc` (eg. `@salesforce/eslint-config-lwc/ssr`) and let developers apply this config for all the components that are required to support SSR.

Reply:
I like this latter approach. We want to reduce friction as much as possible for folks pulling this in, while not breaking the world for everybody else. Disable the rules by default here, and enable through a separate eslint config that can pull in all SSR-related lint rules.

Reply:
There are groups of rules based on where a component will be used. Eg components targeting our offline mobile apps have additional unique rules. Configs sound like a good solution to this. 

cc @khawkins

Reply:
I'll have these rules as part of a separate `ssr` config. These were just carried over from the other repo.